File: RealEstate-Source-Code-v6.0/AdminPanel-Laravel10/app/Http/Controllers/api/v1/AuthController.php
<?php

namespace App\Http\Controllers\api\v1;

use Illuminate\Http\Request;
use App\Models\User;
use Illuminate\Support\Facades\Auth;
use Illuminate\Support\Facades\Validator;

class AuthController extends BaseAPIController
{
    public function login(Request $request)
    {
        $login = $request->validate([
            'username' => 'required|string',
            'password' => 'required|string'
        ]);

        if (!Auth::attempt($login)) {
            return response(['error' => ['message' => 'Invalid login credentials.']]);
            return $this->responseJSON(['error' => ['message' => 'Invalid login credentials!']], 400);
        }

        Auth::user()->tokens()->delete();
        $plainTextToken = Auth::user()->createToken(time())->plainTextToken;

        return $this->responseJSON(['user' => Auth::user(), 'token' => $plainTextToken]);
    }

    public function register(Request $request)
    {
        $input = $request->all();
        $validator = Validator::make($input, [
            'firstName' => 'required',
            'lastName' => 'required',
            'email' => 'required|email',
            'username' => 'required',
            'password' => 'required',
        ]);
        if ($validator->fails()) {
            return $this->responseJSON(['error' => ['message' => 'Validation error, please check fields!']], 400);
        }
        if (User::where('username', $request->username)->exists()) {
            return $this->responseJSON(['error' => ['message' => 'This username is already taken!']], 400);
        }
        if (User::where('email', $request->email)->exists()) {
            return $this->responseJSON(['error' => ['message' => 'This email is already taken!']], 400);
        }
        $user = User::create([
            'first_name' => $request->firstName,
            'last_name' => $request->lastName,
            'email' => $request->email,
            'username' => $request->username,
            'password' => bcrypt($request->password),
            'image_name' => 'default_profile_picture.png'
        ]);
        $success['user'] = $user;
        $user->tokens()->delete();
        $success['token'] = $user->createToken(time())->plainTextToken;
        return $this->responseJSON(['success' => $success], 200);
    }
}

File: RealEstate-Source-Code-v6.0/AdminPanel-NetCore/Controllers/AuthController.cs
using System;
using System.Collections.Generic;
using System.IdentityModel.Tokens.Jwt;
using System.Linq;
using System.Security.Claims;
using System.Text;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Options;
using Microsoft.IdentityModel.Tokens;
using RealEstate.Entities;
using RealEstate.Helpers;
using RealEstate.Models;

namespace RealEstate.Controllers
{
    [Authorize]
    [Route("api/v1/auth")]
    [ApiController]
    public class AuthController : BaseController
    {
        private readonly RealEstateContext _context;
        private readonly Helpers.AppSettings _appSettings;

        public AuthController(RealEstateContext context, IOptions<Helpers.AppSettings> appSettings)
        {
            _context = context;
            _appSettings = appSettings.Value;
        }

        [AllowAnonymous]
        [HttpPost("login")]
        public IActionResult Authenticate(LoginRequestModel requestModel)
        {
            var user = _context.User.SingleOrDefault(x => x.Username == requestModel.Username);

            // check if username exists
            if (user == null)
                return Error("Username or password is incorrect");

            // check if password is correct
            if (!AuthenticationHelper.VerifyPasswordHash(requestModel.Password, user.PasswordHash, user.PasswordSalt))
                return Error("Username or password is incorrect");

            var tokenHandler = new JwtSecurityTokenHandler();
            var key = Encoding.ASCII.GetBytes(_appSettings.Secret);
            var tokenDescriptor = new SecurityTokenDescriptor
            {
                Subject = new ClaimsIdentity(new Claim[]
                {
                    new Claim(ClaimTypes.NameIdentifier, user.Id.ToString()),
                    new Claim(ClaimTypes.Name, user.FirstName.ToString()),
                    new Claim(ClaimTypes.Surname, user.LastName.ToString()),
                    new Claim(ClaimTypes.Role, user.IsAdmin ? "Admin" : user.IsAgent ? "Agent" : "User")
                }),
                Expires = DateTime.UtcNow.AddYears(1),
                SigningCredentials = new SigningCredentials(new SymmetricSecurityKey(key), SecurityAlgorithms.HmacSha256Signature)
            };

            var token = tokenHandler.CreateToken(tokenDescriptor);
            var tokenString = tokenHandler.WriteToken(token);

            // return basic user info (without password) and token to store client side
            return Ok(new
            {
                Token = tokenString,
                user
            });
        }

        [AllowAnonymous]
        [HttpPost("register")]
        public async Task<IActionResult> Create(RegisterRequestModel requestModel)
        {
            if (_context.User.Any(x => x.Username == requestModel.Username))
                return Error("Username \"" + requestModel.Username + "\" is already taken");

            if (_context.User.Any(x => x.Email == requestModel.Email))
                return Error("Email \"" + requestModel.Email + "\" is already taken");

            byte[] passwordHash, passwordSalt;
            AuthenticationHelper.CreatePasswordHash(requestModel.Password, out passwordHash, out passwordSalt);

            var user = new User
            {
                FirstName = requestModel.FirstName,
                LastName = requestModel.LastName,
                IsAdmin = false,
                CreatedDate = DateTime.Now,

                Email = requestModel.Email,
                Username = requestModel.Username,
                PasswordHash = passwordHash,
                PasswordSalt = passwordSalt
            };

            _context.User.Add(user);
            await _context.SaveChangesAsync();

            return Ok();
        }

    }
}